src/components/Order/Stepper.jsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import Box from "@mui/material/Box";
+import Stepper from "@mui/material/Stepper";
+import Step from "@mui/material/Step";
+import StepLabel from "@mui/material/StepLabel";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import useStep from "./useStep";
+import {containerStyle,stepContentBox,buttonContainer,
+         spacer,backButton,completeText} from "./Stepper.styles";
+
+import OrderSummary from "./steps/OrderSummary";
+import CustomerDetails from "./steps/CustomerDetails";
+import PaymentDetails from "./steps/PaymentDetails";
+import OrderComplete from "./steps/OrderComplete";
+
+const stepComponents = [
+    <OrderSummary />,
+    <CustomerDetails />,
+    <PaymentDetails />,
+    <OrderComplete />
+  ];
+  
+const renderStepContent = (step) => stepComponents[step] || null;
+
+const steps = [
+  "Order Summary",
+  "Customer Details",
+  "Payment Details",
+  "Order Complete"
+];
+
+export default function HorizontalLinearStepper() {
+  const { activeStep, next, back } = useStep(steps.length);
+
+  return (
+    <Box sx={containerStyle}>
+      <Stepper activeStep={activeStep}>
+        {steps.map((label) => (
+          <Step key={label}>
+            <StepLabel>{label}</StepLabel>
+          </Step>
+        ))}
+      </Stepper>
+
+      <Box sx={stepContentBox}>{renderStepContent(activeStep)}</Box>
+
```

Comment:
This is a bad practice, because this function will create every time that the component renders (a new object in memory). 

You Don't really need this function so just render your components one by one